python/cudf/cudf/core/column/column.py

```diff
@@ -2032,117 +2037,74 @@ def as_column(
             return ColumnBase.from_scalar(arbitrary, length)
 
     elif hasattr(arbitrary, "__array_interface__"):
-        # CUDF assumes values are always contiguous
         desc = arbitrary.__array_interface__
-        shape = desc["shape"]
-        arb_dtype = np.dtype(desc["typestr"])
+        check_invalid_array(desc["shape"], np.dtype(desc["typestr"]))
+
         # CUDF assumes values are always contiguous
-        if len(shape) > 1:
-            raise ValueError("Data must be 1-dimensional")
+        arbitrary = np.asarray(arbitrary, order="C")
 
-        arbitrary = np.asarray(arbitrary)
+        if arbitrary.ndim == 0:
+            # TODO: Or treat as scalar?
+            arbitrary = arbitrary[np.newaxis]
 
-        # Handle case that `arbitrary` elements are cupy arrays
-        if (
-            shape
-            and shape[0]
-            and hasattr(arbitrary[0], "__cuda_array_interface__")
-        ):
+        if arbitrary.dtype.kind in "OSU":
+            # Handle case that `arbitrary` elements are cupy arrays
+            if len(arbitrary) > 0 and hasattr(
+                arbitrary[0], "__cuda_array_interface__"
+            ):
+                return as_column(
```

Comment:
Correct. I had added this since I thought there was a unit test that would hit this branch. I am not longer encountering that after running the test suite again so I'll actually remove it for now